src/views/jobManager/jobManagerView.ts

```diff
@@ -127,6 +129,76 @@ export class JobManagerView implements TreeDataProvider<any> {
         }
       }),
 
+      vscode.commands.registerCommand(`vscode-db2i.jobManager.editSelfCodes`, async (node?: SQLJobItem) => {
+        const id = node ? node.label as string : undefined;
+        let selected = id ? JobManager.getJob(id) : JobManager.getSelection();
+        if (selected) {
+          try {
+            const currentSelfCodes = selected.job.options.selfcodes;
+            const selfCodeItems: SelfCodesQuickPickItem[] = selfCodesMap.map(
+              (code) => new SelfCodesQuickPickItem(code)
+            );
+            const currentSelfCodeItems: SelfCodesQuickPickItem[] =
+              selfCodeItems.filter((item) =>
+                currentSelfCodes ? currentSelfCodes.includes(item.label) : false
+              );
+
+            const quickPick = vscode.window.createQuickPick();
+            quickPick.title = `Select SELFCODES`;
+            quickPick.canSelectMany = true;
+            quickPick.matchOnDetail = true;
+            quickPick.items = [
+              {
+                kind: vscode.QuickPickItemKind.Separator,
+                label: "Currently selected SELFCODES",
+              },
+              ...currentSelfCodeItems,
+              {
+                kind: vscode.QuickPickItemKind.Separator,
+                label: "All Available SELFCODES",
+              },
+              ...selfCodeItems.filter((item) =>
+                currentSelfCodeItems
+                  ? !currentSelfCodeItems.includes(item)
+                  : true
+              ),
+            ];
+
+            quickPick.selectedItems = currentSelfCodeItems;
+
+            quickPick.onDidAccept(async () => {
+              const selections = quickPick.selectedItems;
+              // SET SYSIBMADM.SELFCODES = SYSIBMADM.VALIDATE_SELF('-514, -204, -501, +30, -199');
+              if (selections) {
+                const codes: string[] = selections.map((code) => code.label);
+                setSelfCodes(codes);
```

Comment:
Here we should be calling an instance method on the selected job (`selected`) to set the SELF codes.